### PR TITLE
Add ValueTaskFactory helpers

### PR DESCRIPTION
**Pull request by tmat**
`ValueTask.FromResult` and `ValueTask.CompletedTask` are only available in net5.0. Adds a helper class that fills in the gap.

### CODE REVIEW

**Comment by tmat** [comment]
@sharwell @CyrusNajmabadi @jasonmalinowski

File: src/Compilers/Core/Portable/InternalUtilities/ValueTaskFactory.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    /// <summary>
+    /// Implements <see cref="ValueTask{TResult}"/> members that are only available in .NET 5.
+    /// </summary>
+    internal static class ValueTaskFactory
+    {
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
+        public static ValueTask<T> FromResult<T>(T result)
+            => new(result);
+#pragma warning restore
```
Comment (sharwell) [inline src/Compilers/Core/Portable/InternalUtilities/ValueTaskFactory.cs:17]: 💡 Use `[SuppressMessage]` similar to `SpecializedTasks`

```suggestion
        [SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "This is a ValueTask wrapper, not an asynchronous method.")]
        public static ValueTask<T> FromResult<T>(T result)
            => new(result);
```

**Review by sharwell — APPROVED** [review]

**Review by sharwell — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4)

**Comment by jcouv** [comment]
There's some legitimate CI errors.

**Comment by tmat** [comment]
@dotnet/roslyn-compiler

**Review by jaredpar — APPROVED** [review]

**Comment by tmat** [comment]
@jaredpar Thanks!

**Pull request merged**